cli/internal/fs/path.go

```diff
@@ -151,6 +162,19 @@ func IofsRelativePath(fsysRoot string, absolutePath string) (string, error) {
 	return filepath.Rel(fsysRoot, absolutePath)
 }
 
+// GetTempDir returns the absolute path of a directory with the given name
+// under the system's default temp directory location
+func GetTempDir(subDir string) AbsolutePath {
+	return AbsolutePath(os.TempDir()).Join(subDir)
+}
+
+// GetTurboDataDir returns a directory outside of the repo
+// where turbo can store data files related to turbo.
+func GetTurboDataDir() AbsolutePath {
+	dataHome := UnsafeToAbsolutePath(xdg.DataHome)
```

Comment:
`Unsafe` has a long, pre-rust history! See, for example, `sun.misc.Unsafe` from Java-land, that underpins every high-performance Java framework. Mostly it just means unchecked, trust me. 

But I see your point. My only problem with direct casts is that they are difficult to find. Maybe `AbsolutePathFromUpstream` or something like that?

Reply:
`AbsolutePathFromUpstream` seems fine. (If renaming, my only goal for this is to remove "Unsafe" from the name so that when I see it I don't go, "oh no, did I do something wrong?")

Reply:
`AbsolutePathFromUpstream` now implemented.